ocs/pkg/service/v0/users.go

```diff
@@ -267,6 +277,53 @@ func (o Ocs) DeleteUser(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	t, err := o.mintTokenForUser(r.Context(), account)
+	if err != nil {
+		render.Render(w,r, response.ErrRender(data.MetaServerError.StatusCode, errors.Wrap(err, "could not mint token").Error()))
+		return
+	}
+
+	ctx := metadata.AppendToOutgoingContext(r.Context(), token.TokenHeader, t)
+
+	homeResp, err := o.revaClient.GetHome(ctx, &provider.GetHomeRequest{} )
+	if err != nil {
+		render.Render(w,r, response.ErrRender(data.MetaServerError.StatusCode, errors.Wrap(err, "could not get home").Error()))
+		return
+	}
+
+	statResp, err := o.revaClient.Stat(ctx, &provider.StatRequest{
+		Ref: &provider.Reference {
+			Spec: &provider.Reference_Path{
+				Path: homeResp.Path,
+			},
+		},
+	})
+	if err != nil {
+		render.Render(w,r, response.ErrRender(data.MetaServerError.StatusCode, errors.Wrap(err, "could not stat home").Error()))
+		return
+	}
+
+	if statResp.Status.Code == rpcv1beta1.Code_CODE_OK {
+		delReq := &provider.DeleteRequest{
+			Ref: &provider.Reference {
+				Spec: &provider.Reference_Id{
+					Id: statResp.Info.Id,
+				},
+			},
+		}
+
+		_, err = o.revaClient.Delete(ctx, delReq)
+		if err != nil {
```

Comment:
you also need to check if `delResp.Status.Code == rpcv1beta1.Code_CODE_OK` again ... sorrd cs3 / grpc is pretty verbose. the `err` code is only indicating if the transport request was successful. we always need to check if the response.Status.Code is OK as well. Distinguishing transport from application logic allows changing the transport layer in the future ... eg using an event bus ...